agreement.md

```diff
@@ -0,0 +1,151 @@
+# Aragon Network Agreement
+
+## Section 0. Background
+The Aragon Network is a Decentralized Autonomous Organization that is building a dispute resolution protocol and other smart contract infrastructure.
+
+The Aragon Network is governed by Aragon Network Token (ANT) holders, who supported the initial creation of the network and under this Agreement will continue to guide its evolution through established governance processes.
+
+This Agreement exists to enumerate the shared mission, principles, and governance guidelines that bind ANT holders in their collaborative effort to create value for the Aragon Network.
+
+## Section 1. RFC 2119 compliance
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED",  "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119.
+
+## Section 2. Immutable guidelines
+The guidelines in this section SHALL NOT be amended. If any other guidelines in this Agreement conflict with these immutable guidelines, the immutable guidelines SHALL take priority for enforcement purposes. These guidelines apply to all proposals submitted to the Aragon Network organization.
+
+1. Proposals MUST be governed by this Agreement, or a future version of this Agreement as modified within the bounds of the Agreement.
+
+2. Disputes over a proposal relying on this Agreement MUST be resolved by Aragon Court.
+
+3. Proposals MUST NOT disproportionately benefit a majority of ANT holders voting on a proposal over the minority.
```

Comment:
Could "disproportionately benefit" become a matter of debate? Can we imagine scenarios where opinions differ on whether a particular proposal is "disproportionately beneficial" to the majority? Without concrete measurement/quantification, how would such a conflict get resolved? If this is specifically about proposals to transfer financial assets from the minority to the majority, would it make sense to say it explicitly?

Reply:
> Could "disproportionately benefit" become a matter of debate? Can we imagine scenarios where opinions differ on whether a particular proposal is "disproportionately beneficial" to the majority? Without concrete measurement/quantification, how would such a conflict get resolved? Aragon Court?

Yes, per the guideline 2.2 (immediately preceding this one) all disputes under this Agreement are to be resolved in Aragon Court.

> If this is specifically about proposals to transfer financial assets from the minority to the majority, would it make sense to say it explicitly?

A 51% attacker might try to take over the network to expropriate wealth from the minority. They could also be after Aragon Court for various reasons: maybe they plan to attack an organization relying on Aragon Court, or maybe they have taken a large short position on ANJ and want to destroy confidence in the court. This guideline is worded the way it is to try and "ban" all possible types of proposals that could be put forward to disproportionately benefit the majority over the minority of voters, without needing to anticipate specific types of proposals attackers might come up with to weasel their way to success.

Reply:
Understood. I realize that a decision mechanism must be implemented, that majority rule is the most obvious option, and that the resulting attack vectors must be addressed somehow. But what about proposals that "disproportionately benefit" a minority over the majority? I think even a cursory look at conventional politics shows that one cannot assume the majority will always vote in their own best interest; indeed, the majority often votes in the interest of an ideologically dominant minority. One might even say that political systems are typically "hacked" and "attacked" by well-organized minorities (elites), not majorities (often politically immobile, disorganized). Does it make sense to raise the following question: why try and ban all proposals that "disproportionately benefit" a majority over the minority, but not the other way around? I think this highlights the ambiguity of "disproportionately benefit" which, as I understand it, would have to be measured and determined by the Aragon Court case-by-case. Opening a can of worms here, for sure, so if the implications of this problem are too complex and impractical to be considered for the Agreement, the discussion can be moved to the Aragon Forum instead.

Reply:
Your points are well taken @mlphresearch! Thank you.

In the Aragon Network whitepaper this guideline was suggested:

> Proposals must be made in good faith with the intention to improve the Network's operational efficiency, quality, or breadth of service, and benefit all ANT holders **in equal measure**.

(emphasis mine)

Would something like this be more along the lines of the type of protection you're looking for?

Reply:
I think so, yes, although "in equal measure" feels simultaneously very precise and difficult to determine. I definitely like "in good faith" and "with the intention to improve the Network"; I'm less sure whether a proposal can benefit everyone **in equal measure**. The "spirit of the Agreement" matters more here than precise measurement; "**disproportionately** benefit" is actually a pretty good way of putting it, but I think it should be phrased/applied more universally, i.e. not only as a safeguard against mob rule but as guidance/insurance for the long-term growth/health of the network and the community as a whole.

Reply:
>I'm less sure whether a proposal can benefit everyone in equal measure.

Instead we can say that the proposal must not be zero-sum. It could benefit different ANT holders unequally but it must not harm anyone.